### ISSUE TITLE
Error: No named parameter with the name 'nullOk'

### ISSUE THREAD

**Issue by BartusZak** [larryaasen/upgrader#89]
I'm trying to use this package for my app. Its probably problem related to my `flutter` version with `null safety` support.

```cmd
Launching lib\main.dart on sdk gphone x86 arm in debug mode...
 lib\main.dart
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/upgrader-2.8.2/lib/src/upgrade_messages.dart:76:48: Error: No named parameter with the name 'nullOk'.
      locale = Localizations.localeOf(context, nullOk: true);
                                               ^^^^^^
/C:/flutter/packages/flutter/lib/src/widgets/localizations.dart:413:17: Context: Found this candidate, but the arguments don't match.
  static Locale localeOf(BuildContext context) {

                ^^^^^^^^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/upgrader-2.8.2/lib/src/upgrader.dart:392:48: Error: No named parameter with the name 'nullOk'.

      locale = Localizations.localeOf(context, nullOk: true);
                                               ^^^^^^
/C:/flutter/packages/flutter/lib/src/widgets/localizations.dart:413:17: Context: Found this candidate, but the arguments don't match.
  static Locale localeOf(BuildContext context) {
                ^^^^^^^^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4/lib/src/structs.dart:1111:7: Info: Struct 'ENUMLOGFONTEX' is empty. Support for empty structs is deprecated and will be removed in the next stable version of Dart. Use Opaque instead.
class ENUMLOGFONTEX extends Struct {
      ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4/lib/src/structs.dart:2835:7: Info: Struct 'BLUETOOTH_PIN_INFO' is empty. Support for empty structs is deprecated and will be removed in the next stable version of Dart. Use Opaque instead.
class BLUETOOTH_PIN_INFO extends Struct {

      ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4/lib/src/structs.dart:2960:7: Info: Struct 'EXCEPINFO' is empty. Support for empty structs is deprecated and will be removed in the next stable version of Dart. Use Opaque instead.

class EXCEPINFO extends Struct {}
      ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4/lib/src/structs.dart:2966:7: Info: Struct 'PROPERTYKEY' is empty. Support for empty structs is deprecated and will be removed in the next stable version of Dart. Use Opaque instead.
class PROPERTYKEY extends Struct {}
      ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4/lib/src/structs.dart:2973:7: Info: Struct 'PROPVARIANT' is empty. Support for empty structs is deprecated and will be removed in the next stable version of Dart. Use Opaque instead.
class PROPVARIANT extends Struct {}
      ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4/lib/src/structs.dart:2978:7: Info: Struct 'SAFEARRAY' is empty. Support for empty structs is deprecated and will be removed in the next stable version of Dart. Use Opaque instead.
class SAFEARRAY extends Struct {}
      ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4/lib/src/structs.dart:2985:7: Info: Struct 'CLSID' is empty. Support for empty structs is deprecated and will be removed in the next stable version of Dart. Use Opaque instead.
class CLSID extends Struct {}
      ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4/lib/src/structs.dart:2992:7: Info: Struct 'STATSTG' is empty. Support for empty structs is deprecated and will be removed in the next stable version of Dart. Use Opaque instead.
class STATSTG extends Struct {}

      ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4/lib/src/structs.dart:2999:7: Info: Struct 'NLM_SIMULATED_PROFILE_INFO' is empty. Support for empty structs is deprecated and will be removed in the next stable version of Dart. Use Opaque instead.
class NLM_SIMULATED_PROFILE_INFO extends Struct {}
      ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/ffi-0.1.3/lib/src/utf8.dart:23:7: Info: Struct 'Utf8' is empty. Support for empty structs is deprecated and will be removed in the next stable version of Dart. Use Opaque instead.
class Utf8 extends Struct {

      ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/ffi-0.1.3/lib/src/utf16.dart:16:7: Info: Struct 'Utf16' is empty. Support for empty structs is deprecated and will be removed in the next stable version of Dart. Use Opaque instead.

class Utf16 extends Struct {
      ^
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/ffi-0.1.3/lib/src/allocation.dart:47:33: Info: Support for using non-constant type arguments 'T' in this FFI API is deprecated and will be removed in the next stable version of Dart. Rewrite the code to ensure that type arguments are compile time constants referring to a valid native type.
  final int totalSize = count * sizeOf<T>();
                                ^

FAILURE: Build failed with an exception.

* Where:
Script 'C:\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 991

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command 'C:\flutter\bin\flutter.bat'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
BUILD FAILED in 39s
Exception: Gradle task assembleDebug failed with exit code 1
Exited (sigterm)
```

```yaml
upgrader: ^2.8.2
```

```cmd
flutter --version
Flutter 1.26.0-17.3.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision 4b50ca7f7f (10 days ago) • 2021-02-04 19:44:27 -0800
Engine • revision 2c527d6c7e
Tools • Dart 2.12.0 (build 2.12.0-259.8.beta)
```

**Comment by larryaasen** [comment]
There will be a null safety version of this package available soon that should resolve this issue. It will let you know when it is available.

**Comment by zjamshidi** [comment]
I upgraded the flutter and now my project fails to build:

```
Flutter 1.27.0-2.0.pre.98 • channel master • https://github.com/flutter/flutter.git
Framework • revision 696e4f70d1 (4 days ago) • 2021-02-13 18:31:02 -0500
Engine • revision 1d537824d6
Tools • Dart 2.13.0 (build 2.13.0-30.0.dev)
```

What should I do?

**Comment by larryaasen** [comment]
This package works on stable right now, and other channels without using null safety. There will be a null safety version of this package available soon that should resolve this issue. It will let you know when it is available.

**Comment by zjamshidi** [comment]
Thanks @larryaasen 
Switching to the stable channel solved the build issue for now

**Comment by larryaasen** [comment]
Just published a null safety version of this package as a prerelease. Check it out https://pub.dev/packages/upgrader/versions.

**Comment by rickyreza12** [comment]
when i tried to update into the latest version i got this error 

`Because no versions of requests match >3.3.0 <4.0.0 and requests 3.3.0 depends on shared_preferences ^0.5.2, requests ^3.3.0 requires shared_preferences ^0.5.2.

And because upgrader >=3.0.0-nullsafety.2 depends on shared_preferences ^2.0.3, requests ^3.3.0 is incompatible with upgrader >=3.0.0-nullsafety.2.

So, because mobile7 depends on both requests ^3.3.0 and upgrader ^3.0.0, version solving failed.
pub get failed (1; So, because mobile7 depends on both requests ^3.3.0 and upgrader ^3.0.0, version solving failed.)`

because i need a package that have and old version make the apps didn't work

**Comment by zjamshidi** [comment]
@rickyreza12 you could fix it by overriding dependencies.

**Comment by larryaasen** [comment]
I just released version 2.8.3 to resolve the nullOk issue. See if that resolves the issue for you.
https://pub.dev/packages/upgrader/versions/2.8.3
